packages/medusa/src/models/batch-job.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  DeleteDateColumn,
+  Entity,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from "typeorm"
+import { ulid } from "ulid"
+import { BatchJobStatus, BatchJobType } from "../types/batch-job"
+import { DbAwareColumn, resolveDbType } from "../utils/db-aware-column"
+
+@Entity()
+export class BatchJob {
+  @PrimaryColumn()
+  id: string
+
+  @DbAwareColumn({ type: "enum", enum: BatchJobType })
+  type: BatchJobType
+
+  @DbAwareColumn({ type: "enum", enum: BatchJobStatus })
+  status: BatchJobStatus
+
+  @Column({ type: "varchar", nullable: true })
```

Comment:
nit (non-blocking): I believe TypeORM picks the proper "varying character" type automatically based on the underlying DB driver when using `type: "text"`, so maybe change to that. Doesn't affect anything here but say we wanted to add MySQL in the future, or some other driver, then this might need to be refactored.